### PR TITLE
DDF-3669 Use list and workspace metacards to determine list actions

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
This PR changes the way list actions are obtained. Currently we are passing only the workspace metacard to the `ActionProvider`s to obtain the list actions. This PR updates the list action retrieval code so that it passes a map containing both the workspace metacard _and_ the list metacard to the `ActionProvider`s instead.

This change will make it easier for `ActionProvider` implementors to write simple and correct code. For example, prior to this change, writing an `ActionProvider` for lists that returns actions only when the list is not empty would require the implementor to deal with the list metacard XML. Now that the list is being provided to each `ActionProvider` as a `Metacard`, writing code that deals with the details of the list is easier.

##### Update
Removes the hostname fix now that support for internal/external URLs has been added.

#### Who is reviewing it? 
@Bdthomson 
@Variadicism 
@glenhein 
@dimitry-sherman

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining 

#### How should this be tested? (List steps with links to updated documentation)
Build [this branch](https://github.com/jrnorth/ddf/tree/list-action-provider-test) from my DDF fork and run `install -s mvn:ddf.catalog.core/catalog-core-listactionprovider/2.12.0-SNAPSHOT` in the DDF console. It's an action provider that provides list actions. Create a list, add some items, and verify that the list actions still work.

#### What are the relevant tickets?
[DDF-3669](https://codice.atlassian.net/browse/DDF-3669)

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by jrnorth** [comment]
Rebased to resolve merge conflicts.

**Comment by bdeining** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformations/EmbeddedListMetacardsHandler.java [hunk 1]
```diff
@@ -82,23 +91,24 @@ private void removeExternalListAttributes(Map listMetacardMap) {
...
 
     return actionRegistry
-        .list(workspaceMetacard)
+        .list(listContext)
         .stream()
         .filter(action -> action.getId().startsWith(LIST_ACTION_PREFIX))
         .map(
             action -> {
-              // Work-around for paths being behind VPCs with non-public DNS values
-              final String url = action.getUrl().toString().replaceFirst(host, "");
```
Comment (coyotesqrl) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformations/EmbeddedListMetacardsHandler.java]: ❓ This is no longer necessary?

Comment (jrnorth) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformations/EmbeddedListMetacardsHandler.java]: We don't need this anymore now that we have support for internal/external URLs.

**Comment by jrnorth** [comment]
Rebased to resolve merge conflicts.

**Comment by Bdthomson** [comment]
build now

**Comment by jrnorth** [comment]
Build now

**Comment by Bdthomson** [comment]
Successfully tested functionality, after installing patch I see the actions list.

File: catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/EmbeddedListMetacardsHandlerTest.java [hunk 2]
```diff
@@ -62,7 +69,13 @@
   public void setup() throws MalformedURLException {
     embeddedListMetacardsHandler = new EmbeddedListMetacardsHandler(actionRegistry);
     final Action action = getMockAction();
-    doReturn(singletonList(action)).when(actionRegistry).list(any());
+    doReturn(singletonList(action))
+        .when(actionRegistry)
+        .list(
```
Comment (jrnorth) [inline catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/EmbeddedListMetacardsHandlerTest.java]: I'm seeing an intermittent compilation error here:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile (default-testCompile) on project catalog-ui-search: Compilation failure
[ERROR] /projects/ddf/catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/EmbeddedListMetacardsHandlerTest.java:[74,14] incompatible types: inferred type does not conform to upper bound(s)
[ERROR]     inferred: java.util.Map<? extends java.lang.String,? extends ddf.catalog.data.Metacard>
[ERROR]     upper bound(s): java.util.Map<? extends java.lang.String,? extends capture#1 of ? extends ddf.catalog.data.Metacard>,java.util.Map<? extends java.lang.String,? extends ddf.catalog.data.Metacard>,java.lang.Object
```

Comment (jrnorth) [reply catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/EmbeddedListMetacardsHandlerTest.java]: Qualifying `allOf` with `Matchers.<Map<String, Metacard>>` appears to have fixed it.

**Comment by bdeining** [comment]
build now

**Pull request merged**